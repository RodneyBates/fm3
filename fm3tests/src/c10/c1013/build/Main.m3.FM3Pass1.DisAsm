 ItkEOF()
 ItkRightEnd()
 ItkModuleLtPatch(C4,L1(16_1),[1,0])
 ItkModuleBeginPatch(C734,L1(16_1),[29,2])
 ItkModuleRt(L1(16_1),[1,0])
 ItkDeclScopeRt(L1(16_1))
 ItkOpenScopeRt(L1(16_1))
 ItkStmtListRt(L0(16_0),[30,2])
 ItkStmtListLt(L0(16_0),[30,2])
 ItkOpenDeclListLtPatch(C10,L5(16_5),[5,2])
 ItkOpenDeclListRt(L5(16_5),[5,2])
 ItkProcDefLtPatch(C618,Id7"P1",[25,2])
 ItkProcDefRt(Id7"P1",[27,8])
 ItkDeclScopeRt(L9(16_9))
 ItkOpenScopeRt(L9(16_9))
 ItkStmtListRt(L0(16_0),[27,4])
 ItkStmtListLt(L0(16_0),[27,4])
 ItkOpenDeclListRt(L0(16_0),[26,4])
 ItkOpenDeclListLt(L0(16_0),[26,4])
 ItkOpenScopeLt(L9(16_9))
 ItkDeclScopeLt(L9(16_9))
 ItkProcDefBody([26,2])
 ItkSignatureProperLtPatch(C636,[25,15])
 ItkSignatureProperRt([25,15])
 ItkRaisesSetRt(L0(16_0),[26,2])
 ItkRaisesSetLt(L0(16_0),[26,2])
 ItkResultTypeAbsent([26,2])
 ItkDeclScopeRt(L8(16_8))
 ItkFormalsListLtPatch(C636,L0(16_0),[25,17])
 ItkFormalsListRt(L0(16_0),[25,17])
 ItkDeclScopeLt(L8(16_8))
 ItkDeclId(L18(16_12),Id7"P1",L-9223372036854775808(16_8000000000000000),[25,12])
 ItkOpenDeclListSep(L4(16_4),[5,2])
 ItkTypeDeclLtPatch(C410,[22,7])
 ItkTypeDeclTypePatch(C418,[22,9])
 ItkTypeDeclRt([22,7])
 ItkDeclScopeRt(L7(16_7))
 ItkObjTypeLtPatch(C418,L0(16_0),L7(16_7),[22,11])
 ItkObjTypeRt(L0(16_0),L7(16_7),[22,11])
 ItkOverrideListRt(L0(16_0),[23,18])
 ItkOverrideListLt(L0(16_0),[23,18])
 ItkMethodDeclListRt(L0(16_0),[23,18])
 ItkMethodDeclListLt(L0(16_0),[23,18])
 ItkFieldDeclListRt(L0(16_0),[23,18])
 ItkFieldDeclListLt(L0(16_0),[23,18])
 ItkDeclScopeLt(L7(16_7))
 ItkBrandAbsent([23,11])
 ItkSupertypeLtPatch(C418,[22,11])
 ItkSupertypeRt([22,11])
 ItkDeclScopeRt(L6(16_6))
 ItkObjTypeLtPatch(C418,L0(16_0),L6(16_6),[22,11])
 ItkObjTypeRt(L0(16_0),L6(16_6),[22,11])
 ItkOverrideListRt(L0(16_0),[22,40])
 ItkOverrideListLt(L0(16_0),[22,40])
 ItkMethodDeclListRt(L0(16_0),[22,40])
 ItkMethodDeclListLt(L0(16_0),[22,40])
 ItkFieldDeclListLtPatch(C429,L1(16_1),[22,18])
 ItkFieldDeclListRt(L1(16_1),[22,18])
 ItkFieldDeclLtPatch(C429,[22,18])
 ItkFieldDeclRt([22,18])
 ItkDeclValAbsent([22,40])
 ItkFieldDeclValue([22,40])
 ItkReservedIdRef(Rid"INTEGER",[22,32])
 ItkFieldDeclType([22,30])
 ItkFieldDeclIdListLtPatch(C429,L1(16_1),[22,18])
 ItkFieldDeclIdListRt(L1(16_1),[22,18])
 ItkDeclId(L16(16_10),Id9"NestedField",L-9223372036854775808(16_8000000000000000),[22,18])
 ItkDeclScopeLt(L6(16_6))
 ItkBrandAbsent([22,11])
 ItkSupertypeAbsent([22,11])
 ItkDeclId(L8(16_8),Id8"T",L181(16_b5),[22,7])
 ItkOpenDeclListSep(L3(16_3),[5,2])
 ItkTypeDeclLtPatch(C277,[15,7])
 ItkTypeDeclTypePatch(C292,[15,10])
 ItkTypeDeclRt([15,7])
 ItkDeclScopeRt(L5(16_5))
 ItkObjTypeLtPatch(C292,L2(16_2),L5(16_5),[15,12])
 ItkObjTypeRt(L2(16_2),L5(16_5),[15,12])
 ItkOverrideListLtPatch(C334,L1(16_1),[18,12])
 ItkOverrideListRt(L1(16_1),[18,12])
 ItkOverrideLtPatch(C334,[19,14])
 ItkOverrideEqualsPatch(C339,[19,17])
 ItkOverrideRt([19,14])
 ItkIdRefAtom(Id7"P1",[19,19])
 ItkOverrideIdAtom(Id4"M1",[19,14])
 ItkMethodDeclListLt(L0(16_0),[17,12])
 ItkMethodDeclListRt(L0(16_0),[17,12])
 ItkFieldDeclListRt(L0(16_0),[17,12])
 ItkFieldDeclListLt(L0(16_0),[17,12])
 ItkDeclScopeLt(L5(16_5))
 ItkBrandExplicitLtPatch(C296,[15,12])
 ItkBrandExplicitRt([15,12])
 ItkIdRefAtom(Id6"Brand",[15,20])
 ItkSupertypeAbsent([15,12])
 ItkDeclId(L8(16_8),Id5"T3",L-9223372036854775808(16_8000000000000000),[15,7])
 ItkOpenDeclListSep(L2(16_2),[5,2])
 ItkTypeDeclLtPatch(C101,[7,7])
 ItkTypeDeclTypePatch(C116,[7,10])
 ItkTypeDeclRt([7,7])
 ItkDeclScopeRt(L3(16_3))
 ItkObjTypeLtPatch(C116,L1(16_1),L3(16_3),[7,12])
 ItkObjTypeRt(L1(16_1),L3(16_3),[7,12])
 ItkOverrideListLt(L0(16_0),[12,12])
 ItkOverrideListRt(L0(16_0),[12,12])
 ItkMethodDeclListLtPatch(C148,L1(16_1),[10,12])
 ItkMethodDeclListRt(L1(16_1),[10,12])
 ItkMethodDeclLtPatch(C148,Id4"M1",[11,14])
 ItkMethodDeclRt(Id4"M1",[11,14])
 ItkSignatureProperLtPatch(C166,[11,17])
 ItkSignatureProperRt([11,17])
 ItkRaisesSetRt(L0(16_0),[12,12])
 ItkRaisesSetLt(L0(16_0),[12,12])
 ItkResultTypeAbsent([12,12])
 ItkDeclScopeRt(L4(16_4))
 ItkFormalsListLtPatch(C166,L0(16_0),[11,19])
 ItkFormalsListRt(L0(16_0),[11,19])
 ItkDeclScopeLt(L4(16_4))
 ItkDeclId(L17(16_11),Id4"M1",L-9223372036854775808(16_8000000000000000),[11,14])
 ItkFieldDeclListRt(L0(16_0),[10,12])
 ItkFieldDeclListLt(L0(16_0),[10,12])
 ItkDeclScopeLt(L3(16_3))
 ItkBrandAnon([8,12])
 ItkSupertypeLtPatch(C116,[7,12])
 ItkSupertypeRt([7,12])
 ItkIdRefAtom(Id2"T1",[7,12])
 ItkDeclId(L8(16_8),Id3"T2",L-9223372036854775808(16_8000000000000000),[7,7])
 ItkOpenDeclListSep(L1(16_1),[5,2])
 ItkTypeDeclLtPatch(C10,[5,7])
 ItkTypeDeclTypePatch(C25,[5,10])
 ItkTypeDeclRt([5,7])
 ItkDeclScopeRt(L2(16_2))
 ItkObjTypeLtPatch(C25,L0(16_0),L2(16_2),[5,12])
 ItkObjTypeRt(L0(16_0),L2(16_2),[5,12])
 ItkOverrideListRt(L0(16_0),[5,19])
 ItkOverrideListLt(L0(16_0),[5,19])
 ItkMethodDeclListRt(L0(16_0),[5,19])
 ItkMethodDeclListLt(L0(16_0),[5,19])
 ItkFieldDeclListRt(L0(16_0),[5,19])
 ItkFieldDeclListLt(L0(16_0),[5,19])
 ItkDeclScopeLt(L2(16_2))
 ItkBrandAbsent([5,12])
 ItkSupertypeAbsent([5,12])
 ItkDeclId(L8(16_8),Id2"T1",L-9223372036854775808(16_8000000000000000),[5,7])
 ItkOpenScopeLt(L1(16_1))
 ItkDeclScopeLt(L1(16_1))
 ItkLeftEnd()
 ItkBOF()
     0
