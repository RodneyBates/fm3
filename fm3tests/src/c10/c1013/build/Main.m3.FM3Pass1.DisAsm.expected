 ItkEOF()
 ItkRightEnd()
 ItkModuleLtPatch(C4,L1(16_1),[1,0])
 ItkModuleBeginPatch(C712,L1(16_1),[29,2])
 ItkModuleRt(L1(16_1),[1,0])
 ItkDeclScopeRt(L1(16_1))
 ItkOpenScopeRt(L1(16_1))
 ItkStmtListRt(L0(16_0),[30,2])
 ItkStmtListLt(L0(16_0),[30,2])
 ItkOpenDeclListLtPatch(C10,L4(16_4),[5,2])
 ItkOpenDeclListRt(L4(16_4),[5,2])
 ItkProcDefLtPatch(C590,Id8"P1",[25,2])
 ItkProcDefRt(Id8"P1",[25,15])
 ItkOpenScopeRt(L8(16_8))
 ItkDeclScopeRt(L9(16_9))
 ItkOpenScopeRt(L9(16_9))
 ItkStmtListRt(L0(16_0),[27,4])
 ItkStmtListLt(L0(16_0),[27,4])
 ItkOpenDeclListRt(L0(16_0),[26,4])
 ItkOpenDeclListLt(L0(16_0),[26,4])
 ItkOpenScopeLt(L9(16_9))
 ItkDeclScopeLt(L9(16_9))
 ItkOpenScopeLt(L8(16_8))
 ItkDeclScopeRt(L8(16_8))
 ItkProcDefBody([26,2])
 ItkSignatureProperLtPatch(C608,[25,15])
 ItkSignatureProperRt([25,15])
 ItkRaisesSetRt(L0(16_0),[26,2])
 ItkRaisesSetLt(L0(16_0),[26,2])
 ItkResultTypeAbsent([26,2])
 ItkFormalsListLtPatch(C608,L0(16_0),[25,17])
 ItkFormalsListRt(L0(16_0),[25,17])
 ItkDeclScopeLt(L8(16_8))
 ItkDeclId(L18(16_12),Id8"P1",L-9223372036854775808(16_8000000000000000),[25,12])
 ItkOpenDeclListSep(L3(16_3),[5,2])
 ItkSkipLtPatch(C402,L1(16_1))
 ItkSkipRt(L1(16_1))
 ItkDeclScopeRt(L7(16_7))
 ItkObjTypeLtPatch(C402,L0(16_0),L7(16_7),[22,11])
 ItkObjTypeRt(L0(16_0),L7(16_7),[22,11])
 ItkOverrideListRt(L0(16_0),[23,18])
 ItkOverrideListLt(L0(16_0),[23,18])
 ItkMethodDeclListRt(L0(16_0),[23,18])
 ItkMethodDeclListLt(L0(16_0),[23,18])
 ItkFieldDeclListRt(L0(16_0),[23,18])
 ItkFieldDeclListLt(L0(16_0),[23,18])
 ItkDeclScopeLt(L7(16_7))
 ItkBrandAbsent([23,11])
 ItkSupertypeLtPatch(C402,[22,11])
 ItkSupertypeRt([22,11])
 ItkDeclScopeRt(L6(16_6))
 ItkObjTypeLtPatch(C402,L0(16_0),L6(16_6),[22,11])
 ItkObjTypeRt(L0(16_0),L6(16_6),[22,11])
 ItkOverrideListRt(L0(16_0),[22,40])
 ItkOverrideListLt(L0(16_0),[22,40])
 ItkMethodDeclListRt(L0(16_0),[22,40])
 ItkMethodDeclListLt(L0(16_0),[22,40])
 ItkFieldDeclListLtPatch(C409,L1(16_1),[22,18])
 ItkFieldDeclListRt(L1(16_1),[22,18])
 ItkFieldDeclLtPatch(C409,[22,18])
 ItkFieldDeclRt([22,18])
 ItkDeclValAbsent([22,40])
 ItkFieldDeclValue([22,30])
 ItkReservedIdRef(Rid"INTEGER",[22,32])
 ItkFieldDeclType([22,18])
 ItkFieldDeclIdListLtPatch(C409,L1(16_1),[22,18])
 ItkFieldDeclIdListRt(L1(16_1),[22,18])
 ItkDeclId(L16(16_10),Id9"NestedField",L-9223372036854775808(16_8000000000000000),[22,18])
 ItkDeclScopeLt(L6(16_6))
 ItkBrandAbsent([22,11])
 ItkOpenDeclListSep(L3(16_3),[5,2])
 ItkTypeDeclLtPatch(C273,[15,7])
 ItkTypeDeclTypePatch(C288,[15,10])
 ItkTypeDeclRt([15,7])
 ItkDeclScopeRt(L5(16_5))
 ItkObjTypeLtPatch(C288,L2(16_2),L5(16_5),[15,12])
 ItkObjTypeRt(L2(16_2),L5(16_5),[15,12])
 ItkOverrideListLtPatch(C326,L1(16_1),[18,12])
 ItkOverrideListRt(L1(16_1),[18,12])
 ItkOverrideLtPatch(C326,[19,14])
 ItkOverrideEqualsPatch(C331,[19,17])
 ItkOverrideRt([19,14])
 ItkIdRefAtom(Id8"P1",[19,19])
 ItkOverrideIdAtom(Id5"M1",[19,14])
 ItkMethodDeclListLt(L0(16_0),[17,12])
 ItkMethodDeclListRt(L0(16_0),[17,12])
 ItkFieldDeclListRt(L0(16_0),[17,12])
 ItkFieldDeclListLt(L0(16_0),[17,12])
 ItkDeclScopeLt(L5(16_5))
 ItkBrandExplicitLtPatch(C288,[15,12])
 ItkBrandExplicitRt([15,12])
 ItkIdRefAtom(Id7"Brand",[15,20])
 ItkDeclId(L8(16_8),Id6"T3",L-9223372036854775808(16_8000000000000000),[15,7])
 ItkOpenDeclListSep(L2(16_2),[5,2])
 ItkTypeDeclLtPatch(C97,[7,7])
 ItkTypeDeclTypePatch(C112,[7,10])
 ItkTypeDeclRt([7,7])
 ItkDeclScopeRt(L3(16_3))
 ItkObjTypeLtPatch(C112,L1(16_1),L3(16_3),[7,12])
 ItkObjTypeRt(L1(16_1),L3(16_3),[7,12])
 ItkOverrideListLt(L0(16_0),[12,12])
 ItkOverrideListRt(L0(16_0),[12,12])
 ItkMethodDeclListLtPatch(C144,L1(16_1),[10,12])
 ItkMethodDeclListRt(L1(16_1),[10,12])
 ItkDeclScopeRt(L4(16_4))
 ItkMethodDeclLtPatch(C144,Id5"M1",[11,14])
 ItkMethodDeclRt(Id5"M1",[11,14])
 ItkSignatureProperLtPatch(C162,[11,17])
 ItkSignatureProperRt([11,17])
 ItkRaisesSetRt(L0(16_0),[12,12])
 ItkRaisesSetLt(L0(16_0),[12,12])
 ItkResultTypeAbsent([12,12])
 ItkFormalsListLtPatch(C162,L0(16_0),[11,19])
 ItkFormalsListRt(L0(16_0),[11,19])
 ItkDeclScopeLt(L4(16_4))
 ItkDeclId(L17(16_11),Id5"M1",L-9223372036854775808(16_8000000000000000),[11,14])
 ItkFieldDeclListRt(L0(16_0),[10,12])
 ItkFieldDeclListLt(L0(16_0),[10,12])
 ItkDeclScopeLt(L3(16_3))
 ItkBrandAnon([8,12])
 ItkSupertypeLtPatch(C112,[7,12])
 ItkSupertypeRt([7,12])
 ItkIdRefAtom(Id3"T1",[7,12])
 ItkDeclId(L8(16_8),Id4"T2",L-9223372036854775808(16_8000000000000000),[7,7])
 ItkOpenDeclListSep(L1(16_1),[5,2])
 ItkTypeDeclLtPatch(C10,[5,7])
 ItkTypeDeclTypePatch(C25,[5,10])
 ItkTypeDeclRt([5,7])
 ItkDeclScopeRt(L2(16_2))
 ItkObjTypeLtPatch(C25,L0(16_0),L2(16_2),[5,12])
 ItkObjTypeRt(L0(16_0),L2(16_2),[5,12])
 ItkOverrideListRt(L0(16_0),[5,19])
 ItkOverrideListLt(L0(16_0),[5,19])
 ItkMethodDeclListRt(L0(16_0),[5,19])
 ItkMethodDeclListLt(L0(16_0),[5,19])
 ItkFieldDeclListRt(L0(16_0),[5,19])
 ItkFieldDeclListLt(L0(16_0),[5,19])
 ItkDeclScopeLt(L2(16_2))
 ItkBrandAbsent([5,12])
 ItkDeclId(L8(16_8),Id3"T1",L-9223372036854775808(16_8000000000000000),[5,7])
 ItkOpenScopeLt(L1(16_1))
 ItkDeclScopeLt(L1(16_1))
 ItkLeftEnd()
 ItkBOF()
     0
